benchmarks/cmake/AddBenchmark.cmake

```diff
@@ -0,0 +1,200 @@
+        "${options}"
+        "${one_value_args}"
+        "${multi_value_args}"
+        ${ARGN}
+    )
+
+    set(ISPC_DST_DIR "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/CMakeFiles/ispc/")
+    file(TO_NATIVE_PATH "${ISPC_DST_DIR}" ISPC_DST_DIR)
+    file(MAKE_DIRECTORY ${ISPC_DST_DIR})
+
+    string(FIND ${BENCHMARKS_ISPC_TARGETS} "," MULTI_TARGET)
+
+    foreach(ISPC_SRC_FILE ${ADD_ISPC_SOURCES})
+        set(ISPC_TARGET_HEADERS "")
+        set(ISPC_TARGET_OBJS "")
+
+        # Full path to source file
+        get_filename_component(SRC_LOCATION "${ISPC_SRC_FILE}" ABSOLUTE "${CMAKE_CURRENT_SOURCE_DIR}")
+
+        # Construct names for header and object files
+        string(REPLACE ".ispc" "${CMAKE_CXX_OUTPUT_EXTENSION}" ISPC_OBJ "${ISPC_SRC_FILE}")
+        set(ISPC_OBJ "${ISPC_DST_DIR}${ISPC_OBJ}")
+        string(REPLACE ".ispc" "_ispc.h" ISPC_HEADER "${ISPC_SRC_FILE}")
+        set(ISPC_HEADER "${ISPC_DST_DIR}${ISPC_HEADER}")
+        list(APPEND ISPC_TARGET_OBJS "${ISPC_OBJ}")
+        list(APPEND ISPC_TARGET_HEADERS "${ISPC_HEADER}")
+
+        # Collect list of expected outputs in case of multiple targets
+        if(${MULTI_TARGET} GREATER -1)
+            foreach (ISPC_TARGET ${ISPC_KNOWN_TARGETS})
+                string(FIND ${BENCHMARKS_ISPC_TARGETS} ${ISPC_TARGET} FOUND_TARGET)
+                if(${FOUND_TARGET} GREATER -1)
+                    set(OUTPUT_TARGET ${ISPC_TARGET})
+                    if (${ISPC_TARGET} STREQUAL "avx1")
+                        set(OUTPUT_TARGET "avx")
+                    endif()
+                    string(REPLACE ".ispc" "_${OUTPUT_TARGET}${CMAKE_CXX_OUTPUT_EXTENSION}" ISPC_TARGET_OBJ ${ISPC_SRC_FILE})
+                    set(ISPC_TARGET_OBJ ${ISPC_DST_DIR}${ISPC_TARGET_OBJ})
+                    list(APPEND ISPC_TARGET_OBJS ${ISPC_TARGET_OBJ})
+                    string(REPLACE ".ispc" "_ispc_${OUTPUT_TARGET}.h" ISPC_TARGET_HEADER ${ISPC_SRC_FILE})
+                    set(ISPC_TARGET_HEADER ${ISPC_DST_DIR}${ISPC_TARGET_HEADER})
+                    list(APPEND ISPC_TARGET_HEADERS ${ISPC_TARGET_HEADER})
+                endif()
+            endforeach()
+        endif()
+
+        if(UNIX)
+            set(ISPC_PIC "--pic")
+        endif()
+
+        # Passing space separate string yields escaped spaces.
+        # So convert to a list and then use generator expression, i.e. "$<JOIN:${FLAGS},;>"
+        separate_arguments(FLAGS NATIVE_COMMAND ${BENCHMARKS_ISPC_FLAGS})
+
+        add_custom_command(
+            OUTPUT ${ISPC_TARGET_OBJS} ${ISPC_TARGET_HEADERS}
+            COMMENT "Compiling ${ISPC_SRC_FILE} for ${BENCHMARKS_ISPC_TARGETS} target(s)"
+            COMMAND           ${ISPC_EXECUTABLE} ${SRC_LOCATION} -o ${ISPC_OBJ} -h ${ISPC_HEADER} --arch=${ISPC_ARCH} --target=${BENCHMARKS_ISPC_TARGETS} ${ISPC_PIC} "$<JOIN:${FLAGS},;>"
+            DEPENDS ${ISPC_EXECUTABLE}
+            DEPENDS ${ISPC_SRC_FILE}
+            COMMAND_EXPAND_LISTS
+        )
+        if(MSVC)
```

Comment:
Please change the body of MSVC condition to:
`        if(MSVC)
            # Add .ispc file to VS solution.
	    target_sources(${ADD_ISPC_TARGET} PUBLIC ${SRC_LOCATION})
	    # Group .ispc files inside Visual Studio
            source_group("ISPC" FILES ${SRC_LOCATION})
	    # Group benchmarks in "Benchmarks" folder
	    set_target_properties(${ADD_ISPC_TARGET} PROPERTIES FOLDER "Benchmarks")
        endif()`

Reply:
Thanks! Done.